app/connectors/aws/S3EventParser.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.aws
+
+import javax.inject.Inject
+
+import model.{FileUploadEvent, Message, S3ObjectLocation}
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
+import play.api.libs.json._
+import services._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class S3EventParser @Inject()(implicit ec: ExecutionContext) extends MessageParser {
+
+  case class S3EventNotification(records: Seq[S3EventNotificationRecord])
+
+  case class S3EventNotificationRecord(
+    eventVersion: String,
+    eventSource: String,
+    awsRegion: String,
+    eventTime: String,
+    eventName: String,
+    s3: S3Details)
```

Comment:
Very minor, could we rename this property to s3Details? As in other places I think the code is using s3 for an S3Client object.